Wuxiliang/matrix one local

qingxinhome opened the pull request:
**What type of PR is this?**

- [ ] API-change
- [x] BUG
- [ ] Improvement
- [ ] Documentation
- [ ] Feature
- [ ] Test and CI
- [ ] Code Refactoring

**Which issue(s) this PR fixes:**

issue #1659

**What this PR does / why we need it:**

Not Available

**Special notes for your reviewer:**

Not Available

**Additional documentation (e.g. design docs, usage docs, etc.):**

Not Available

m-schen reviewed (changes requested):
from the code, (--1) will be prune as -1 ?

please check the performance of SQL
`select * from tbl where 1 = (--1)` 
is right or wrong @qingxinhome

qingxinhome commented:
> from the code, (--1) will be prune as -1 ?
> 
> please check the performance of SQL `select * from tbl where 1 = (--1)` is right or wrong @qingxinhome Yes, it exist the problem that - 1 will be pruned to - 1, I have fixed it and tested it

Yes, it exist the problem that - 1 will be pruned to - 1, I have fixed it and tested it
